src/wallet/wallet.cpp

```diff
@@ -1625,13 +1625,21 @@ static void ApproximateBestSubset(vector<pair<CAmount, pair<const CWalletTx*,uns
                             nBest = nTotal;
                             vfBest = vfIncluded;
                         }
-                        nTotal -= vValue[i].first;
```

Comment:
This (along with `&& !fReachedTarget` is a new, different change to the algorithm than the post-processing step initially proposed.

I think we should separate these, the postprocessing to remove extraneous outputs below is obviously correct, but I'm not sure about this.